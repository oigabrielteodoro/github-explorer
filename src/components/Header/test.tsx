import React, { ReactNode } from 'react'
import { render, screen } from '@testing-library/react'

import { Header } from '.'

type LinkProps = {
  children: ReactNode
}

jest.mock('react-router-dom', () => ({
  Link: ({ children }: LinkProps) => children,
  useLocation() {
    return {
      pathname: '/dasda',
    }
  },
}))

describe('<Header />', () => {
  it('should render correctly', () => {
    const { container } = render(<Header />)

    expect(screen.getByRole('img')).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })
})
