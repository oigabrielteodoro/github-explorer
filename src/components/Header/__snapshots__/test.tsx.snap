// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 105rem;
  margin: 4.0rem auto 0;
}

.c0 a {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 500;
  -webkit-transition: 0.1s ease-in-out;
  transition: 0.1s ease-in-out;
  color: #a8a8b3;
  margin-left: auto;
}

.c0 a:hover {
  -webkit-filter: brightness(0.8);
  filter: brightness(0.8);
}

.c0 a svg {
  margin-right: 0.6rem;
}

<header
  class="c0"
>
  <img
    alt="Logo"
    src="/img/logo.svg"
  />
</header>
`;
